AmplifyPlugins/Core/AWSPluginsCore/Model/GraphQLRequest/GraphQLRequest+Model.swift

```diff
@@ -112,25 +114,43 @@ protocol ModelGraphQLRequestFactory {
 extension GraphQLRequest: ModelGraphQLRequestFactory {
 
     public static func create<M: Model>(_ model: M) -> GraphQLRequest<M> {
-        return mutation(of: model, type: .create)
+        let modelType = ModelRegistry.modelType(from: model.modelName) ?? Swift.type(of: model)
+        return mutation(of: model, modelSchema: modelType.schema, type: .create)
+    }
+
+    public static func update<M: Model>(_ model: M,
+                                        where predicate: QueryPredicate? = nil) -> GraphQLRequest<M> {
+        let modelType = ModelRegistry.modelType(from: model.modelName) ?? Swift.type(of: model)
+        return mutation(of: model, modelSchema: modelType.schema, where: predicate, type: .update)
```

Comment:
Here and elsewhere: My preference would be to delegate less-specific overrides to the more-specific ones, as in:

```suggestion
        let modelType = ModelRegistry.modelType(from: model.modelName) ?? Swift.type(of: model)
        let modelSchema = modelType.schema
        return update(model, modelSchema: schema, where: predicate)
```

More DRY and easier to follow the code paths.